### PR TITLE
11556: Add StyleCop and fix violations

### PR DESCRIPTION
**Pull request by PhysicsJon**
[AB#11554](https://buyingcatalog.visualstudio.com/c5f97979-5b03-4d10-ba8d-871d0526b408/_workitems/edit/11554)
[AB#11556](https://buyingcatalog.visualstudio.com/c5f97979-5b03-4d10-ba8d-871d0526b408/_workitems/edit/11556)

### CODE REVIEW

File: src/.editorconfig [hunk 1]
```diff
@@ -59,6 +59,10 @@ dotnet_naming_symbols.constant_fields.required_modifiers          = const
 ###############################
 # C# Coding Conventions       #
 ###############################
+
+# IDE0060: Remove unused parameter
+dotnet_diagnostic.IDE0060.severity = none
```
Comment (bjss-chris) [inline src/.editorconfig:64]: Seems like an odd one to suppress 😃

**Review by bjss-chris — COMMENTED** [review]
Just one comment but otherwise looks good.

File: src/.editorconfig (hunk 1 above)
Comment (PhysicsJon) [reply src/.editorconfig:64]: I forgot to un-supress that one

Comment (PhysicsJon) [reply src/.editorconfig]: Unsuppressed and fixed

**Review by bjss-chris — APPROVED** [review]
👍

**Pull request merged**